diff --git a/third_party/xsimd/include/xsimd/arch/xsimd_wasm.hpp b/third_party/xsimd/include/xsimd/arch/xsimd_wasm.hpp
index 9e437dfb1a67adb279f41e838cf5aab9ea437f4e..b657bbd6b439fa5ca6cba80c4b6aae2d3f3b2b80 100644
--- a/third_party/xsimd/include/xsimd/arch/xsimd_wasm.hpp
+++ b/third_party/xsimd/include/xsimd/arch/xsimd_wasm.hpp
@@ -13,11 +13,11 @@
 #ifndef XSIMD_WASM_HPP
 #define XSIMD_WASM_HPP
 
-#include <type_traits>
-
 #include "../types/xsimd_wasm_register.hpp"
 #include "./common/xsimd_common_cast.hpp"
 
+#include <type_traits>
+
 namespace xsimd
 {
     template <typename T, class A, bool... Values>
