diff --git a/third_party/xsimd/include/xsimd/arch/xsimd_vsx.hpp b/third_party/xsimd/include/xsimd/arch/xsimd_vsx.hpp
index bf4c08308702b93ea57f09d8d90d5fd10d7690bd..4a184136dc6f1fd8067ac974e833a307bc69e8e5 100644
--- a/third_party/xsimd/include/xsimd/arch/xsimd_vsx.hpp
+++ b/third_party/xsimd/include/xsimd/arch/xsimd_vsx.hpp
@@ -12,14 +12,15 @@
 #ifndef XSIMD_VSX_HPP
 #define XSIMD_VSX_HPP
 
-#include <complex>
-#include <type_traits>
-
+#include "../types/xsimd_batch_fwd.hpp"
 #include "../types/xsimd_vsx_register.hpp"
 #include "./common/xsimd_common_cast.hpp"
 
 #include <endian.h>
 
+#include <complex>
+#include <type_traits>
+
 namespace xsimd
 {
     template <typename T, class A, bool... Values>
@@ -33,6 +34,37 @@ namespace xsimd
 
     namespace kernel
     {
+        // builtin_t<T> - the scalar type as it would be used for a vector intrinsic
+        // VSX vector intrinsics do not support long, unsigned long, and char
+        // The builtin<T> definition can be used to map the incoming
+        // type to the right one to be used with the intrinsics.
+        template <typename T>
+        struct builtin_scalar
+        {
+            using type = T;
+        };
+
+        template <>
+        struct builtin_scalar<unsigned long>
+        {
+            using type = unsigned long long;
+        };
+
+        template <>
+        struct builtin_scalar<long>
+        {
+            using type = long long;
+        };
+
+        template <>
+        struct builtin_scalar<char>
+        {
+            using type = typename std::conditional<std::is_signed<char>::value, signed char, unsigned char>::type;
+        };
+
+        template <typename T>
+        using builtin_t = typename builtin_scalar<T>::type;
+
         template <class A, class T>
         XSIMD_INLINE batch<T, A> avg(batch<T, A> const&, batch<T, A> const&, requires_arch<common>) noexcept;
         template <class A, class T>
@@ -218,7 +250,7 @@ namespace xsimd
         template <class A, class T, class = std::enable_if_t<std::is_scalar<T>::value>>
         XSIMD_INLINE batch<T, A> broadcast(T val, requires_arch<vsx>) noexcept
         {
-            return vec_splats(val);
+            return vec_splats(static_cast<builtin_t<T>>(val));
         }
 
         // ceil
@@ -421,18 +453,18 @@ namespace xsimd
             return ~vec_cmpeq(self.data, self.data);
         }
 
-        // load_aligned
+        // load_unaligned
         template <class A, class T, class = std::enable_if_t<std::is_scalar<T>::value>>
-        XSIMD_INLINE batch<T, A> load_aligned(T const* mem, convert<T>, requires_arch<vsx>) noexcept
+        XSIMD_INLINE batch<T, A> load_unaligned(T const* mem, convert<T>, requires_arch<vsx>) noexcept
         {
-            return vec_ld(0, reinterpret_cast<const typename batch<T, A>::register_type*>(mem));
+            return (typename batch<T, A>::register_type)vec_xl(0, (builtin_t<T>*)mem);
         }
 
-        // load_unaligned
+        // load_aligned
         template <class A, class T, class = std::enable_if_t<std::is_scalar<T>::value>>
-        XSIMD_INLINE batch<T, A> load_unaligned(T const* mem, convert<T>, requires_arch<vsx>) noexcept
+        XSIMD_INLINE batch<T, A> load_aligned(T const* mem, convert<T>, requires_arch<vsx>) noexcept
         {
-            return vec_vsx_ld(0, (typename batch<T, A>::register_type const*)mem);
+            return load_unaligned<A>(mem, kernel::convert<T> {}, vsx {});
         }
 
         // load_complex
@@ -605,8 +637,18 @@ namespace xsimd
         }
 
         // round
-        template <class A, class T, class = std::enable_if_t<std::is_floating_point<T>::value>>
-        XSIMD_INLINE batch<T, A> round(batch<T, A> const& self, requires_arch<vsx>) noexcept
+
+        // vec_round exists also for float vectors but is mapped to vrfin instruction which uses the wrong rounding mode
+#if defined __has_builtin && __has_builtin(__builtin_vsx_xvrspi)
+        template <class A>
+        XSIMD_INLINE batch<float, A> round(batch<float, A> const& self, requires_arch<vsx>) noexcept
+        {
+            return __builtin_vsx_xvrspi(self.data);
+        }
+#endif
+        // For double vectors vec_round uses xvrdpi which does the right thing
+        template <class A>
+        XSIMD_INLINE batch<double, A> round(batch<double, A> const& self, requires_arch<vsx>) noexcept
         {
             return vec_round(self.data);
         }
@@ -748,14 +790,14 @@ namespace xsimd
         template <class A, class T, class = std::enable_if_t<std::is_scalar<T>::value>>
         XSIMD_INLINE void store_aligned(T* mem, batch<T, A> const& self, requires_arch<vsx>) noexcept
         {
-            return vec_st(self.data, 0, reinterpret_cast<typename batch<T, A>::register_type*>(mem));
+            vec_xst((typename batch<T, A>::register_type)self.data, 0, (builtin_t<T>*)mem);
         }
 
         // store_unaligned
         template <class A, class T, class = std::enable_if_t<std::is_scalar<T>::value>>
         XSIMD_INLINE void store_unaligned(T* mem, batch<T, A> const& self, requires_arch<vsx>) noexcept
         {
-            return vec_vsx_st(self.data, 0, reinterpret_cast<typename batch<T, A>::register_type*>(mem));
+            store_aligned<A>(mem, self, vsx {});
         }
 
         // sub
