diff --git a/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp b/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp
--- a/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp
+++ b/xpcom/reflect/xptcall/md/unix/xptcstubs_riscv64.cpp
@@ -137,11 +137,11 @@
       "_ZN14nsXPTCStubBase7Stub"#n"Ev: \n\t"                                        \
       ".else  \n\t"                                                                 \
       ".err   \"stub number "#n" >= 1000 not yet supported\"\n"                     \
       ".endif \n\t"                                                                 \
       "li      t0, "#n" \n\t"                                                       \
-      "j       SharedStub \n"                                                       \
+      "tail    SharedStub \n"                                                       \
       ".if "#n" < 10\n\t"                                                           \
       ".size   _ZN14nsXPTCStubBase5Stub"#n"Ev,.-_ZN14nsXPTCStubBase5Stub"#n"Ev\n\t" \
       ".elseif "#n" < 100\n\t"                                                      \
       ".size   _ZN14nsXPTCStubBase6Stub"#n"Ev,.-_ZN14nsXPTCStubBase6Stub"#n"Ev\n\t" \
       ".else\n\t"                                                                   \

