diff --git a/mozglue/build/mips.cpp b/mozglue/build/mips.cpp
--- a/mozglue/build/mips.cpp
+++ b/mozglue/build/mips.cpp
@@ -4,10 +4,12 @@
 
 /* compile-time and runtime tests for whether to use MIPS-specific extensions */
 
 #include "mips.h"
 
+#include "mozilla/Attributes.h"
+
 #include <stdio.h>
 #include <string.h>
 
 enum {
   MIPS_FLAG_LOONGSON3 = 1,
@@ -34,8 +36,8 @@
   return (mips_cpu_flags & MIPS_FLAG_LOONGSON3) != 0;
 }
 
 namespace mozilla {
 namespace mips_private {
-bool isLoongson3 = check_loongson3();
+MOZ_RUNINIT bool isLoongson3 = check_loongson3();
 }  // namespace mips_private
 }  // namespace mozilla
diff --git a/mozglue/build/ppc.cpp b/mozglue/build/ppc.cpp
--- a/mozglue/build/ppc.cpp
+++ b/mozglue/build/ppc.cpp
@@ -5,10 +5,12 @@
 /* compile-time and runtime tests for whether to use Power ISA-specific
  * extensions */
 
 #include "ppc.h"
 
+#include "mozilla/Attributes.h"
+
 #if defined(XP_LINUX)
 // Use the getauxval() function if available.
 // ARCH_3_00 wasn't defined until glibc 2.23, so include just in case.
 #  include <sys/auxv.h>
 #  ifndef PPC_FEATURE2_ARCH_3_00
@@ -51,10 +53,10 @@
   return cpu_flags;
 }
 
 namespace mozilla {
 namespace ppc_private {
-bool vmx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VMX);
-bool vsx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX);
-bool vsx3_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX3);
+MOZ_RUNINIT bool vmx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VMX);
+MOZ_RUNINIT bool vsx_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX);
+MOZ_RUNINIT bool vsx3_enabled = !!(get_ppc_cpu_flags() & PPC_FLAG_VSX3);
 }  // namespace ppc_private
 }  // namespace mozilla

