diff --git a/third_party/xsimd/include/xsimd/types/xsimd_traits.hpp b/third_party/xsimd/include/xsimd/types/xsimd_traits.hpp
index 34b47c7ee305362fab35e7554e6e78ffcbdfb085..0f7485ae4b5348101c12dea07c18a8ce74b3fda3 100644
--- a/third_party/xsimd/include/xsimd/types/xsimd_traits.hpp
+++ b/third_party/xsimd/include/xsimd/types/xsimd_traits.hpp
@@ -12,10 +12,17 @@
 #ifndef XSIMD_TRAITS_HPP
 #define XSIMD_TRAITS_HPP
 
+#include <complex>
 #include <cstdint>
 #include <type_traits>
 
-#include "xsimd_batch.hpp"
+#ifdef XSIMD_ENABLE_XTL_COMPLEX
+#include <xtl/xcomplex.hpp>
+#endif
+
+#include "../config/xsimd_config.hpp"
+#include "./xsimd_batch_fwd.hpp"
+#include "./xsimd_utils.hpp"
 
 /**
  * high level type traits
@@ -54,7 +61,7 @@ namespace xsimd
             static constexpr size_t size = 1;
         };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
         template <class T>
         constexpr size_t simd_traits_impl<T, false>::size;
 #endif
@@ -67,7 +74,7 @@ namespace xsimd
             static constexpr size_t size = type::size;
         };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
         template <class T>
         constexpr size_t simd_traits_impl<T, true>::size;
 #endif
@@ -133,7 +140,7 @@ namespace xsimd
         static constexpr size_t size = simd_traits<type>::size;
     };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
     template <class T>
     constexpr size_t revert_simd_traits<T>::size;
 #endif
@@ -145,7 +152,7 @@ namespace xsimd
         static constexpr size_t size = batch<T>::size;
     };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
     template <class T>
     constexpr size_t revert_simd_traits<batch<T>>::size;
 #endif
@@ -251,7 +258,7 @@ namespace xsimd
         static constexpr bool is_complex = detail::is_complex<T>::value; ///< True if T is complex or a batch of complex values.
     };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
     template <class T>
     constexpr bool batch_traits<T>::is_batch;
     template <class T>
@@ -274,7 +281,7 @@ namespace xsimd
         static constexpr bool is_complex = detail::is_complex<T>::value;
     };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
     template <class T, class A>
     constexpr bool batch_traits<batch<T, A>>::is_batch;
     template <class T, class A>
@@ -297,7 +304,7 @@ namespace xsimd
         static constexpr bool is_complex = false;
     };
 
-#if __cplusplus < 201703L
+#if XSIMD_CPP_VERSION < 201703L
     template <class T, class A>
     constexpr bool batch_traits<batch_bool<T, A>>::is_batch;
     template <class T, class A>
@@ -397,53 +404,6 @@ namespace xsimd
 
     template <class T>
     using mask_type_t = typename mask_type<T>::type;
-
-    namespace detail
-    {
-        template <typename T>
-        struct widen
-        {
-            using type = std::make_signed_t<typename widen<std::make_unsigned_t<T>>::type>;
-        };
-
-        template <>
-        struct widen<uint32_t>
-        {
-            using type = uint64_t;
-        };
-        template <>
-        struct widen<uint16_t>
-        {
-            using type = uint32_t;
-        };
-        template <>
-        struct widen<uint8_t>
-        {
-            using type = uint16_t;
-        };
-        template <>
-        struct widen<int32_t>
-        {
-            using type = int64_t;
-        };
-        template <>
-        struct widen<int16_t>
-        {
-            using type = int32_t;
-        };
-        template <>
-        struct widen<int8_t>
-        {
-            using type = int16_t;
-        };
-        template <>
-        struct widen<float>
-        {
-            using type = double;
-        };
-    }
-    template <typename T>
-    using widen_t = typename detail::widen<T>::type;
 }
 
 #endif
