diff --git a/third_party/xsimd/include/xsimd/types/xsimd_utils.hpp b/third_party/xsimd/include/xsimd/types/xsimd_utils.hpp
index aa11b90dbdafb1bbaa1305e9ffd95c14637ea718..5dbab8551c7c9f0051afc1b7323f7784f3c768ec 100644
--- a/third_party/xsimd/include/xsimd/types/xsimd_utils.hpp
+++ b/third_party/xsimd/include/xsimd/types/xsimd_utils.hpp
@@ -20,18 +20,13 @@
 #include <utility>
 
 #ifdef XSIMD_ENABLE_XTL_COMPLEX
-#include "xtl/xcomplex.hpp"
+#include <xtl/xcomplex.hpp>
 #endif
 
+#include "./xsimd_batch_fwd.hpp"
+
 namespace xsimd
 {
-
-    template <class T, class A>
-    class batch;
-
-    template <class T, class A>
-    class batch_bool;
-
     /**************
      * index      *
      **************/
@@ -211,9 +206,25 @@ namespace xsimd
              * enabling / disabling metafunctions *
              **************************************/
 
+            template <class T>
+            using enable_arithmetic_t = std::enable_if_t<std::is_arithmetic<T>::value, int>;
+
+            /// Enable signed integral or floating point
+            template <class T>
+            using enable_signed_numeral_t = std::enable_if_t<std::is_signed<T>::value, int>;
+
+            template <class T>
+            using enable_floating_point_t = std::enable_if_t<std::is_floating_point<T>::value, int>;
+
             template <class T>
             using enable_integral_t = std::enable_if_t<std::is_integral<T>::value, int>;
 
+            template <class T>
+            using enable_signed_integral_t = std::enable_if_t<std::is_integral<T>::value && std::is_signed<T>::value, int>;
+
+            template <class T>
+            using enable_unsigned_integral_t = std::enable_if_t<std::is_integral<T>::value && std::is_unsigned<T>::value, int>;
+
             template <class T, size_t S>
             using enable_sized_signed_t = std::enable_if_t<std::is_integral<T>::value && std::is_signed<T>::value && sizeof(T) == S, int>;
 
