diff --git a/third_party/xsimd/include/xsimd/types/xsimd_batch_constant.hpp b/third_party/xsimd/include/xsimd/types/xsimd_batch_constant.hpp
index 0f5d4a9787b57c6840906898d896e75030c8b7e7..edecee3c3b29926329a999c73a3b3777f01c7dbf 100644
--- a/third_party/xsimd/include/xsimd/types/xsimd_batch_constant.hpp
+++ b/third_party/xsimd/include/xsimd/types/xsimd_batch_constant.hpp
@@ -12,13 +12,14 @@
 #ifndef XSIMD_BATCH_CONSTANT_HPP
 #define XSIMD_BATCH_CONSTANT_HPP
 
+#include "../config/xsimd_config.hpp"
+#include "./xsimd_batch.hpp"
+#include "./xsimd_utils.hpp"
+
 #include <cstddef>
 #include <functional>
 #include <utility>
 
-#include "./xsimd_batch.hpp"
-#include "./xsimd_utils.hpp"
-
 namespace xsimd
 {
     /**
@@ -408,7 +409,7 @@ namespace xsimd
             return {};
         }
 
-#if __cplusplus >= 202002L
+#if XSIMD_CPP_VERSION >= 202002L
         template <std::array Arr, class A, std::size_t... Is>
         XSIMD_INLINE constexpr batch_constant<typename decltype(Arr)::value_type, A, Arr[Is]...>
         make_batch_constant(std::index_sequence<Is...>) noexcept
@@ -431,7 +432,7 @@ namespace xsimd
             return {};
         }
 
-#if __cplusplus >= 202002L
+#if XSIMD_CPP_VERSION >= 202002L
         template <typename T, std::array Arr, class A, std::size_t... Is>
         XSIMD_INLINE constexpr batch_bool_constant<T, A, Arr[Is]...>
         make_batch_bool_constant(std::index_sequence<Is...>) noexcept
@@ -497,7 +498,7 @@ namespace xsimd
         return {};
     }
 
-#if __cplusplus >= 202002L
+#if XSIMD_CPP_VERSION >= 202002L
     /**
      * @brief Build a @c batch_constant from a std::array (C++20)
      *
@@ -524,7 +525,7 @@ namespace xsimd
         return {};
     }
 
-#if __cplusplus >= 202002L
+#if XSIMD_CPP_VERSION >= 202002L
     /**
      * @brief Build a @c batch_constant from a std::array of boolean (C++20)
      *
