diff --git a/third_party/xsimd/include/xsimd/arch/common/xsimd_common_memory.hpp b/third_party/xsimd/include/xsimd/arch/common/xsimd_common_memory.hpp
index 709497ec220832ea504b124640b87070497cbff0..7a1ed73a3a7d152ad21b98d1e7bfe12f1c21b874 100644
--- a/third_party/xsimd/include/xsimd/arch/common/xsimd_common_memory.hpp
+++ b/third_party/xsimd/include/xsimd/arch/common/xsimd_common_memory.hpp
@@ -12,13 +12,13 @@
 #ifndef XSIMD_COMMON_MEMORY_HPP
 #define XSIMD_COMMON_MEMORY_HPP
 
+#include "../../types/xsimd_batch_constant.hpp"
+#include "./xsimd_common_details.hpp"
+
 #include <algorithm>
 #include <array>
 #include <complex>
 
-#include "../../types/xsimd_batch_constant.hpp"
-#include "./xsimd_common_details.hpp"
-
 namespace xsimd
 {
     template <typename T, class A, T... Values>
@@ -223,7 +223,8 @@ namespace xsimd
         template <class A, size_t I, class T>
         XSIMD_INLINE typename batch<std::complex<T>, A>::value_type get(batch<std::complex<T>, A> const& self, ::xsimd::index<I>, requires_arch<common>) noexcept
         {
-            alignas(A::alignment()) T buffer[batch<std::complex<T>, A>::size];
+            using value_type = typename batch<std::complex<T>, A>::value_type;
+            alignas(A::alignment()) value_type buffer[batch<std::complex<T>, A>::size];
             self.store_aligned(&buffer[0]);
             return buffer[I];
         }
