diff --git a/third_party/xsimd/include/xsimd/arch/common/xsimd_common_cast.hpp b/third_party/xsimd/include/xsimd/arch/common/xsimd_common_cast.hpp
index 1226c887cb9396616037f8a74a10ae8dea8719b6..95753babd1134ee921b6ca7fd0519ecec86ebacf 100644
--- a/third_party/xsimd/include/xsimd/arch/common/xsimd_common_cast.hpp
+++ b/third_party/xsimd/include/xsimd/arch/common/xsimd_common_cast.hpp
@@ -12,7 +12,10 @@
 #ifndef XSIMD_COMMON_CAST_HPP
 #define XSIMD_COMMON_CAST_HPP
 
-#include "../../types/xsimd_traits.hpp"
+#include "../../config/xsimd_macros.hpp"
+#include "../../utils/xsimd_type_traits.hpp"
+
+#include <array>
 
 namespace xsimd
 {
